src/utils/map.ts

```diff
@@ -2,32 +2,27 @@
 
 module Plottable {
 export module Utils {
-  /**
-   * An associative array that can be keyed by anything (inc objects).
-   * Uses pointer equality checks which is why this works.
-   * This power has a price: everything is linear time since it is actually backed by an array...
-   */
   export class Map<K, V> {
-    private _keyValuePairs: any[][] = [];
+    private _keyValuePairs: { key: K; value: V; }[] = [];
 
     /**
-     * Set a new key/value pair in the store.
+     * Set a new key/value pair in the Map.
      *
-     * @param {K} key Key to set in the store
-     * @param {V} value Value to set in the store
-     * @return {boolean} True if key already in store, false otherwise
+     * @param {K} key Key to set in the Map
+     * @param {V} value Value to set in the Map
+     * @return {boolean} True if key already in Map, false otherwise
      */
     public set(key: K, value: V) {
       if (key !== key) {
         throw new Error("NaN may not be used as a key to the Map");
       }
       for (var i = 0; i < this._keyValuePairs.length; i++) {
-        if (this._keyValuePairs[i][0] === key) {
-          this._keyValuePairs[i][1] = value;
+        if (this._keyValuePairs[i].key === key) {
```

Comment:
Shouldn't this be `this._keyValuePairs[i][key]`?  Wouldn't this look up the wrong entry in the pairs?